### PR TITLE
test(notifications): Add notification services e2e mocks

### PR DESCRIPTION
**Pull request by Prithpal-Sooriya**
## **Description**

This adds mocks for the notification services when running e2e tests.

This also unblocks the notifications team from starting e2e tests for our flows.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25065?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/controllers/push-platform-notifications/mocks/mockResponse.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    endpoint: '',
+    p256dh: '',
+    auth: '',
+    applicationPubKey: '',
+  },
+};
+
+export function getMockCreateFCMRegistrationTokenResponse() {
+  return {
+    url: /https:\/\/fcmregistrations\.googleapis\.com\/v1\/projects\/.*/u,
```
[the remark this answers is not in the document]
Comment (Prithpal-Sooriya) [reply app/scripts/controllers/push-platform-notifications/mocks/mockResponse.ts]: Regex is now bounded.

File: app/scripts/controllers/metamask-notifications/mocks/mockServices.ts
Comment (matteoscurati) [inline app/scripts/controllers/metamask-notifications/mocks/mockServices.ts:1]: Cool, so you mocked all the services we need, right?

**Review by matteoscurati — APPROVED** [review]
Ok, great PR that enables us to manage the e2e tests for notifications!

File: app/scripts/controllers/metamask-notifications/mocks/mockServices.ts
Comment (Prithpal-Sooriya) [reply app/scripts/controllers/metamask-notifications/mocks/mockServices.ts:1]: Yep exactly.

These `mockResponses` just allow us to port the mocks cross different mocking systems (e.g. `nock` for unit tests and `mockttp` for e2e tests.

File: test/e2e/tests/notifications/mocks.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  mockAPICall(server, getMockBatchDeleteTriggersResponse());
+  mockAPICall(server, getMockListNotificationsResponse());
+  mockAPICall(server, getMockMarkNotificationsAsReadResponse());
+
+  // Push Notifications
+  mockAPICall(server, getMockRetrievePushNotificationLinksResponse());
+  mockAPICall(server, getMockUpdatePushNotificationLinksResponse());
+  mockAPICall(server, getMockCreateFCMRegistrationTokenResponse());
+  mockAPICall(server, getMockDeleteFCMRegistrationTokenResponse());
+}
```
Comment (Prithpal-Sooriya) [inline test/e2e/tests/notifications/mocks.ts:58]: This is the new file for our e2e mocks.

This folder will also be where our e2e tests for our notifications flow will live.

File: test/env.js [hunk 3]
```diff
@@ -11,3 +11,5 @@ process.env.CONTENTFUL_ACCESS_TOKEN = 'MOCK_ACCESS_TOKEN';
 process.env.TRIGGERS_SERVICE_URL = 'https://mock-test-triggers-api.metamask.io';
 process.env.NOTIFICATIONS_SERVICE_URL =
   'https://mock-test-notifications-api.metamask.io';
+process.env.PUSH_NOTIFICATIONS_SERVICE_URL =
```
Comment (Prithpal-Sooriya) [inline test/env.js:14]: This is for jest mocking.

**Review by dovydas55 — APPROVED** [review]

**Pull request merged**